src/discord.ts

```diff
@@ -123,71 +122,94 @@ export const sendLiveAnnouncement = async (streamInfo: StreamInfo) => {
         content: `${onlineAnnouncementPrefix}${Discord.Util.escapeMarkdown(
-          streamInfo.user_name
-        )} is now live on Twitch! https://twitch.tv/${streamInfo.user_name}`,
+          info.streamer_info.name
+        )} is now live on ${info.streaming_service.name}! ${info.streaming_service.base_url}${info.streamer_info.name}`,
         embed,
       });
     } else {
       message = await announcementsChannel.send({
         content: `${onlineAnnouncementPrefix}${Discord.Util.escapeMarkdown(
-          streamInfo.user_name
-        )} is now live on Twitch! https://twitch.tv/${streamInfo.user_name}`,
+          info.streamer_info.name
+        )} is now live on ${info.streaming_service.name}! ${info.streaming_service.base_url}${info.streamer_info.name}`,
         embed,
       });
     }
 
     await DiscordAnnouncementModel.updateOne(
-      { memberId: streamInfo.user_id },
+      { memberId: info.streamer_info.id },
       {
-        memberId: streamInfo.user_id,
+        memberId: info.streamer_info.id,
         messageId: message.id,
-        streamId: streamInfo.id,
-        category: category.name,
+        streamId: info.id,
+        category: info.category_name,
       },
       { upsert: true }
     );
   }
 };
-
-export const sendOfflineAnnouncement = async (member_id: string) => {
-  const video = await fetchVideoByUserId(member_id);
-
-  if (!video) {
-    return;
-  }
-
+export const sendOfflineAnnouncement = async (announcement: StreamAnnouncementInfo) => {
   const saved_message = await DiscordAnnouncementModel.findOne({
-    memberId: member_id,
+    memberId: announcement.streamer_info.id,
   });
 
   if (!saved_message) {
     return;
   }
-
   const message = await announcementsChannel.messages.fetch(
     `${saved_message.messageId}`
   );
-
-  const user = await UserManager.getUserById(member_id);
-
+  let footer = "Finished streaming";
+  if (announcement.archivedStream) {
+    footer += ` • Streamed for ${announcement.archivedStream.duration}`;
+  }
   const embed = buildDiscordEmbed(
-    false,
-    user.name,
-    user.display_name,
-    user.logo,
-    video.title,
-    video.thumbnail_url,
-    `Finished streaming • Streamed for ${video.duration}`,
+    announcement.online,
+    announcement.streamer_info.name,
+    announcement.streamer_info.display_name,
+    announcement.streamer_info.avatar_url,
+    announcement.title,
+    announcement.thumbnail_url,
+    footer,
     saved_message.category,
-    video.id
+    announcement.streaming_service,
+    announcement.archivedStream?.id
   );
 
   await message.edit({
-    content: `${user.display_name} was online!`,
+    content: `${announcement.streamer_info.display_name} was online!`,
     embed,
   });
 
-  await DiscordAnnouncementModel.deleteOne({ memberId: member_id });
+  await DiscordAnnouncementModel.deleteOne({ memberId: announcement.streamer_info.id });
+};
+
+export const sendTwitchOfflineAnnouncement = async (member_id: string) => {
```

Comment:
Is this here just to co-erce the data into the right format? Would this be better as a helper function, rather that specifically attached to the discord class? Called like ... `sendOfflineAnnouncement(prepareTwitchOfflineMessage(member_id))`